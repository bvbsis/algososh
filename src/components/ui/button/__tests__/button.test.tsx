import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "../button";

describe("Button component :", () => {
  it("should be rendered with text", () => {
    const tree = renderer.create(<Button text="Нажми меня" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be rendered WITHOUT text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be rendered disabled", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be rendered with loader", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call the callback by click", () => {
    const mockCallback = jest.fn();
    mockCallback.mockImplementation((text) => {
      console.log(text);
    });
    render(<Button text={"Нажми меня"} onClick={() => mockCallback('test')} />);
    const button = screen.getByText("Нажми меня");
    fireEvent.click(button);
    expect(mockCallback).toHaveBeenCalledWith('test');
  });
});
