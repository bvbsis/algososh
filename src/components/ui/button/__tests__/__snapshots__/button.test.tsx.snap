// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component : should be rendered WITHOUT text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component : should be rendered disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component : should be rendered with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component : should be rendered with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Нажми меня
  </p>
</button>
`;
